docs/moz-addon-manager.md

```diff
@@ -96,94 +96,57 @@ at a URL like `localhost:3000`.
+@@ -664,12 +664,11 @@ var LightWeightThemeWebInstaller = {
+   },
+ 
+   _isAllowed(principal) {
+-    if (!principal || !principal.URI || !principal.URI.schemeIs("https")) {
++    if (!principal || !principal.URI) {
        return false;
      }
-
--    if (!uri.schemeIs("https")) {
--      return false;
--    }
--
+ 
 -    let pm = Services.perms;
--    return pm.testPermission(uri, "install") == pm.ALLOW_ACTION;
+-    return pm.testPermission(principal.URI, "install") == pm.ALLOW_ACTION;
 +    return true;
-   }
- };
+   },
+ 
+   _shouldShowUndoPrompt(principal) {
 diff --git a/toolkit/mozapps/extensions/AddonManager.jsm b/toolkit/mozapps/extensions/AddonManager.jsm
 --- a/toolkit/mozapps/extensions/AddonManager.jsm
 +++ b/toolkit/mozapps/extensions/AddonManager.jsm
-@@ -64,16 +64,17 @@ var PREF_EM_CHECK_COMPATIBILITY = MOZ_CO
-
- const VALID_TYPES_REGEXP = /^[\w\-]+$/;
-
- const WEBAPI_INSTALL_HOSTS = ["addons.mozilla.org", "testpilot.firefox.com"];
+@@ -57,7 +57,7 @@ const WEBAPI_INSTALL_HOSTS = ["addons.mo
  const WEBAPI_TEST_INSTALL_HOSTS = [
    "addons.allizom.org", "addons-dev.allizom.org",
    "testpilot.stage.mozaws.net", "testpilot.dev.mozaws.net",
-   "example.com",
-+  "localhost",
+-  "example.com",
++  "example.com", "localhost",
  ];
-
+ 
  const URI_XPINSTALL_DIALOG = "chrome://mozapps/content/xpinstall/xpinstallConfirm.xul";
-
- Cu.import("resource://gre/modules/Services.jsm");
- Cu.import("resource://gre/modules/XPCOMUtils.jsm");
- Cu.import("resource://gre/modules/AsyncShutdown.jsm");
-
 diff --git a/toolkit/mozapps/extensions/AddonManagerWebAPI.cpp b/toolkit/mozapps/extensions/AddonManagerWebAPI.cpp
 --- a/toolkit/mozapps/extensions/AddonManagerWebAPI.cpp
 +++ b/toolkit/mozapps/extensions/AddonManagerWebAPI.cpp
-@@ -45,16 +45,17 @@ IsValidHost(const nsACString& host) {
-   // When testing allow access to the developer sites.
-   if (Preferences::GetBool("extensions.webapi.testing", false)) {
-     if (host.LowerCaseEqualsLiteral("addons.allizom.org") ||
-         host.LowerCaseEqualsLiteral("discovery.addons.allizom.org") ||
-         host.LowerCaseEqualsLiteral("addons-dev.allizom.org") ||
+@@ -57,6 +57,7 @@ IsValidHost(const nsACString& host) {
          host.LowerCaseEqualsLiteral("discovery.addons-dev.allizom.org") ||
          host.LowerCaseEqualsLiteral("testpilot.stage.mozaws.net") ||
          host.LowerCaseEqualsLiteral("testpilot.dev.mozaws.net") ||
 +        host.LowerCaseEqualsLiteral("localhost") ||
          host.LowerCaseEqualsLiteral("example.com")) {
        return true;
      }
-   }
-
-   return false;
- }
-
-@@ -64,19 +65,16 @@ bool
- AddonManagerWebAPI::IsValidSite(nsIURI* uri)
- {
-   if (!uri) {
-     return false;
-   }
-
-   bool isSecure;
-   nsresult rv = uri->SchemeIs("https", &isSecure);
--  if (NS_FAILED(rv) || !isSecure) {
--    return false;
--  }
-
-   nsAutoCString host;
-   rv = uri->GetHost(host);
-   if (NS_FAILED(rv)) {
-     return false;
-   }
-
-   return IsValidHost(host);
```

Comment:
should we leave this part in the patch - that this should be removed? I didn't see anything after this info here about removing this piece:

```javascript
@@ -57,6 +57,7 @@ IsValidHost(const nsACString& host) {
          host.LowerCaseEqualsLiteral("discovery.addons-dev.allizom.org") ||
          host.LowerCaseEqualsLiteral("testpilot.stage.mozaws.net") ||
          host.LowerCaseEqualsLiteral("testpilot.dev.mozaws.net") ||
 +        host.LowerCaseEqualsLiteral("localhost") ||
          host.LowerCaseEqualsLiteral("example.com")) {
        return true;
      }
```
see: https://monosnap.com/file/zFkJrRKzqrQfqzi3l9UW5Ny9jY17E2#embed

Reply:
It is already in the code if I am not mistaken

Reply:
ah well wait, you are right actually. I thought we would not need it but I might be wrong.